### PR TITLE
Added integration tests for Repositories.List

### LINKED ISSUE #335: Add integration tests for Repositories.List
I'd like to work on this. To keep API calls to a minimum:
- Repositories.List("", nil) to confirm listing own repositories does not return an error
- Repositories.List("google", nil) to confirm listing other user's repositories does not return an error
- Repositories.List("google", {Sort: "created"}) to confirm options are working

### PR DESCRIPTION
**Pull request by ainsleyc**
Fixes #335

### CODE REVIEW

**Comment by shurcooL** [comment]
Why is this closed? Was it merged?

**Comment by ainsleyc** [comment]
Ah, I saw there were other pulls that needed attention and didn't want to burden the maintainers.

**Comment by shurcooL** [comment]
That's a shame, at a glance, this looked like a positive improvement.

**Comment by willnorris** [comment]
I'll triage appropriately.  No need to close pull requests preemptively just because others are pending as well.  Yes, it may take me (or someone else) a little time to get around to looking at this, but that doesn't mean it's not still valuable.

**Comment by ainsleyc** [comment]
Fair enough. Thanks!

**Comment by gmlewis** [comment]
LGTM :+1: 
@shurcooL - does this LGTY?

**Comment by shurcooL** [comment]
LGTM. Thanks @ainsleyc.

Has anyone tried running this integration test and confirmed it passes? I only reviewed the code, did not run these.

File: tests/integration/repos_test.go [hunk 1]
```diff
@@ -145,3 +145,31 @@ func TestRepositories_EditBranches(t *testing.T) {
...
+	opt := github.RepositoryListOptions{Sort: "created"}
+	repos, _, err := client.Repositories.List("google", &opt)
+	if err != nil {
+		t.Fatalf("Repositories.List('google') with Sort opt returned error: %v", err)
+	} else {
+		for i, repo := range repos {
+			if i > 0 && (*repos[i-1].CreatedAt).Time.Before((*repo.CreatedAt).Time) {
+				t.Fatalf("Repositories.List('google') with default descending Sort returned incorrect order")
+			}
+		}
```
Comment (shurcooL) [inline tests/integration/repos_test.go]: Minor style suggestion (completely optional). There's no need for the for loop to be inside the else clause, you can just pull it out to normal path.

Comment (ainsleyc) [inline tests/integration/repos_test.go]: Done, thanks!

**Comment by gmlewis** [comment]
@shurcooL - I ran this new integration test and it passes. I discovered that some of the TestUsers* integration tests are failing, but these need further investigation. I'll create an issue if one doesn't already exist.

**Comment by gmlewis** [comment]
Integrated in d4aea47.

**Pull request closed without merging**